src/credentials/CHIPCert.h

```diff
@@ -529,6 +530,14 @@ CHIP_ERROR VerifyCertSignature(const ChipCertificateData & cert, const ChipCerti
  */
 CHIP_ERROR ValidateChipRCAC(const ByteSpan & rcac);
 
+/**
+ * Validates a Network (Client) Identity in TLV-encoded form.
+ *
+ * This function parses the certificate, ensures the rigid fields have the values mandated by the
+ * specification, and validates the certificate signature.
```

Comment:
Please provide at least a section name (number is too unstable over time).

Please explain how we trust the public key. We can't just magically trust the public key, given the cert has a "garbage" issuer field. Issuer == Subject always means that it's a self-signed cert, which implies a root cert and the key is not super relevant. This is very unorthodox without a lot of explanation.

Reply:
Discussed offline. Added a reference to the "Wi-F Authentication with Per-Device Credentials" spec section to cover the "why" questions.